sound/soc/sof/control.c

```diff
@@ -405,6 +405,7 @@ int snd_sof_bytes_ext_put(struct snd_kcontrol *kcontrol,
 	struct snd_ctl_tlv header;
```

Comment:
Commit message is rather unclear:

This patch avoids the scontrol persistent storage to be written
over by firmware fail response if the IPC send was not
successful. 

What? Did you mean 
"with the existing code, the configuration is saved in kcontrol cached data even when an IPC fails".


Such situation is possible e.g. if processing
component like EQ responds to IPC with -EBUSY when new
configuration data is passed from user space. The firmware reply
in case of error was copied directly to scontrol data in function
tx_wait_done() and the fail response when re-applied by PM
restore caused another IPC fail.

That last part is unclear. Did you mean "the previous configuration that was saved after the previous IPC failure will be applied by PM restore"? That would be fine, but what is the issue. Why would this cause another IPC to fail? The fail is not a result of the configuration proper but the state of the DSP when the configuration was passed. There is something really fishy about the explanations, sorry.

Reply:
@plbossart @singalsu I think the problem is that when the IPC fails with -EBUSY, the firmware does not write the control data into the mailbox. So what is copied into the control data in the driver when the IPC reply is received is invalid data. This is what causes the very next IPC to the comp to fail during PM restore.

Reply:
I'll improve the description text as well. The sequence of error is a bit complicated. I didn't first realize that the data to stored scontrol can loop via firmware. And that the response will be re-applied as IPC later. The second IPC fail is not even relevant, I'll mention only the first issue.

Reply:
Done, hope it's better now!